### ISSUE TITLE
Warning: Contract code size exceeds 24576 bytes

### ISSUE THREAD

**Issue by barakman** [sc-forks/solidity-coverage#635]
I'd except the following `.solcover.js` to suppress this warning:
```
module.exports = {
    providerOptions: {
        allowUnlimitedContractSize: true
    }
};
```
But it doesn't.

Any idea how to resolve this?

UPDATE:

Second thought, configuring the provider's (runtime) behavior could obviously not yield any impact on this error, which occurs during compilation, so I guess that it is in fact a 'solc' issue rather than a 'solidity-coverage' issue.

Nevertheless, following this compilation warning, I'd expect the execution itself to fail due to the fact that some of the (instrumented) contracts exceed the maximum size. And the execution doesn't fail, **even** if I omit the `allowUnlimitedContractSize` configuration above.

**Comment by cgewecke** [comment]
Hi @barakman :) 

Yes solidity-coverage sets this property to true by default here.  If can be omitted from the config

https://github.com/sc-forks/solidity-coverage/blob/3c0f3a5c7db26e82974873bbf61cf462072a7c6d/lib/api.js#L153-L156

RE: the warnings - the  0.8.x `beta` version suppresses all compiler warnings. It can be used in lieu of v0.7.x without changing anything and its behavioral differences [are documented here][1] 

[1]: https://github.com/sc-forks/solidity-coverage/releases/tag/v0.8.0-beta.0
```
npm install --save-dev solidity-coverage@beta
```

**Comment by barakman** [comment]
Hi @cgewecke :)

Thank you for your quick response.

I should have guessed that SC configures this by default (in fact, I think that it did occur to me some time after I had posted this issue).

I take it that you probably also set the gas limit to `Number.MAX_SAFE_INTEGER` by default...?

Just to be sure, it seems to me that at this point, I can incorporate SC into my project without a `.solcover.js` file to begin with.

Moreover, in contrast with previous SC versions, there's no need to add an additional network in the Truffle configuration file.

So in short, the only thing required at this point (SC 0.7.x) is `plugins: ["solidity-coverage"]`. Is that more or less correct?

Lastly, when is SC 0.8.x due for release (non-beta version)?

Thanks again :)

**Comment by cgewecke** [comment]
> I can incorporate SC into my project without a .solcover.js file to begin with.

Yes, you should be able to if you don't need any special options. The client is auto-configured for gas, etc...

> So in short, the only thing required at this point (SC 0.7.x) is plugins: ["solidity-coverage"]. Is that more or less correct?

If you're using hardhat you just import or require `solidity-coverage` at the top of `hardhat.config.js` (or .ts)

>  Lastly, when is SC 0.8.x due for release (non-beta version)

Soon. It should already be done tbh. I'm going to fix some bugs here today and then see where we're at....

**Comment by barakman** [comment]
Thank you!
